drivers/adodb-mysqli.inc.php

```diff
@@ -157,16 +157,64 @@ function SetTransactionMode($transaction_mode)
 	 * Parameter must be one of the constants listed in mysqli_options().
 	 * @see https://www.php.net/manual/en/mysqli.options.php
 	 *
-	 * @param int    $parameter The parameter to set
-	 * @param string $value     The value of the parameter
+	 * @param string|int $parameter The parameter to set
+	 * @param string|int|array $value The value of the parameter
 	 *
 	 * @return bool
 	 */
 	public function setConnectionParameter($parameter, $value) {
-		if(!is_numeric($parameter)) {
+
+		// Special case for setting SSL cert values.
+		if ("sslcert" === $parameter && is_array($value)) {
+
+			if (true === isset($value["ssl_key"])) {
+				$this->ssl_key = $value["ssl_key"];
+			}
+
+			if (true === isset($value["ssl_cert"])) {
+				$this->ssl_cert = $value["ssl_cert"];
+			}
+
+			if (true === isset($value["ssl_ca"])) {
+				$this->ssl_ca = $value["ssl_ca"];
+			}
+
+			if (true === isset($value["ssl_capath"])) {
+				$this->ssl_capath = $value["ssl_capath"];
+			}
+
+			if (true === isset($value["ssl_cipher"])) {
+				$this->ssl_cipher = $value["ssl_cipher"];
+			}
+
+			return true;
+
+		}
+
+		// Special case for setting the client flag.
+		if ("clientflags" === $parameter && is_numeric($value)) {
+			$this->clientFlags = $value;
+			return true;
```

Comment:
Shouldn't we return `false` if $value is not numeric ? 

With this implementation, ADOConnection::setConnectionParameter() will just blindly set $connectionParameters property to whatever has been given, probably resulting in an Exception to be thrown later in _connect().

https://github.com/ADOdb/ADOdb/blob/1e0e602ed72628f7a6665f5c925e93a80bdbd710/drivers/adodb-mysqli.inc.php#L219-L222

NOTE: This comment also applies to the other special cases with a type check that you added (sslsert, socket, port).